src/tools/rest-backend/service_config.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file
+ *
+ * @brief implementation of the config service class
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include <algorithm>
+#include <regex>
+
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <service.hpp>
+
+namespace kdbrest
+{
+
+namespace service
+{
+
+const std::string REGEX_CONF_KEY = "((?:[a-zA-Z_]+(?:\\.?[a-zA-Z_])+)?)\\.?#_?([0-9]+)\\.?((?:[a-zA-Z0-9_#]+(?:\\.?[a-zA-Z0-9_#])+)?)";
```

Comment:
Yes it is purely for arrays. I tried all combinations that came to my mind with an online regex tester and it seems to work properly. The whole transformation was probably the most complicated stuff...